test/CompactStringVectorTest.cpp

```diff
@@ -137,6 +139,97 @@ TEST(CompactVectorOfStrings, SerializationWithPush) {
   testSerializationWithPush(CompactVectorInt{}, ints);
 }
 
+// Test that a `CompactStringVectorWriter` can be moved, even when writing has
+// already started.
+TEST(CompactVectorOfStrings, MoveWriterWhileWriting1) {
+  auto testSerializationWithPush = [](const auto& v, auto& inputVector) {
+    using V = std::decay_t<decltype(v)>;
+
+    const std::string filename = "_writerTest1029348.dat";
+    {
+      // Move-assign and move-construct before pushing anything.
+      typename V::Writer writer1{filename};
+      auto writer0{std::move(writer1)};
+      writer1 = std::move(writer0);
+
+      std::optional<typename V::Writer> writer2;
+      auto* writer = &writer1;
+      size_t i = 0;
+      for (const auto& s : inputVector) {
+        if (i == 1) {
+          // Move assignment after push.
+          writer0 = std::move(*writer);
+          writer = &writer0;
+        }
+        if (i == 2) {
+          // Move construction after push.
+          writer2.emplace(std::move(*writer));
+          writer = &writer2.value();
+        }
+        writer->push(s.data(), s.size());
+        ++i;
+      }
+    }  // The constructor finishes writing the file.
+
+    V compactVector;
+    ad_utility::serialization::FileReadSerializer ser{filename};
+    ser >> compactVector;
+
+    vectorsEqual(inputVector, compactVector);
+
+    ad_utility::deleteFile(filename);
+  };
+  testSerializationWithPush(CompactVectorChar{}, strings);
+  testSerializationWithPush(CompactVectorInt{}, ints);
+}
+
+// Test the special case of moving into a `CompactStringVectorWriter` that has
+// already been written to.
+TEST(CompactVectorOfStrings, MoveWriterWhileWriting2) {
```

Comment:
Instead of using 1 & 2 suffixes, can you just specify what this test does in the name? I also don't quite see the difference of those two tests from the description alone